src/iccpd/include/mlacp_fsm.h

```diff
@@ -66,6 +69,85 @@ struct Remote_System
     uint32_t node_id;
 };
 
+/****************************************************************
+ * Debug counters to track message sent and received between
+ * MC-LAG peers over ICCP
+ ***************************************************************/
+typedef uint8_t ICCP_DBG_CNTR_DIR_e;
+enum ICCP_DBG_CNTR_DIR_e
+{
+    ICCP_DBG_CNTR_DIR_TX  = 0,
+    ICCP_DBG_CNTR_DIR_RX  = 1,
+    ICCP_DBG_CNTR_DIR_MAX
+};
+
+typedef uint8_t ICCP_DBG_CNTR_STS_e;
+enum ICCP_DBG_CNTR_STS_e
+{
+    ICCP_DBG_CNTR_STS_OK  = 0,
+    ICCP_DBG_CNTR_STS_ERR = 1,     /* Send error or receive processing error*/
+    ICCP_DBG_CNTR_STS_MAX
+};
+
+/* Change MCLAGDCTL_MAX_DBG_COUNTERS if ICCP_DBG_CNTR_MSG_MAX is more than 32 */
+enum ICCP_DBG_CNTR_MSG
+{
+    ICCP_DBG_CNTR_MSG_SYS_CONFIG       = 0,
+    ICCP_DBG_CNTR_MSG_AGGR_CONFIG      = 1,
+    ICCP_DBG_CNTR_MSG_AGGR_STATE       = 2,
+    ICCP_DBG_CNTR_MSG_MAC_INFO         = 3,
+    ICCP_DBG_CNTR_MSG_ARP_INFO         = 4,
+    ICCP_DBG_CNTR_MSG_L2MC_INFO        = 5,
+    ICCP_DBG_CNTR_MSG_PORTCHANNEL_INFO = 6,
+    ICCP_DBG_CNTR_MSG_PEER_LINK_INFO   = 7,
+    ICCP_DBG_CNTR_MSG_HEART_BEAT       = 8,
+    ICCP_DBG_CNTR_MSG_NAK              = 9,
+    ICCP_DBG_CNTR_MSG_SYNC_DATA        = 10,
+    ICCP_DBG_CNTR_MSG_SYNC_REQ         = 11,
+    ICCP_DBG_CNTR_MSG_WARM_BOOT        = 12,
+    ICCP_DBG_CNTR_MSG_IF_UP_ACK        = 13,
+    ICCP_DBG_CNTR_MSG_STP_CONNECT      = 14,
+    ICCP_DBG_CNTR_MSG_STP_DISCONNECT   = 15,
+    ICCP_DBG_CNTR_MSG_STP_SYSTEM_CONFIG = 16,
+    ICCP_DBG_CNTR_MSG_STP_REGION_NAME  = 17,
+    ICCP_DBG_CNTR_MSG_STP_REVISION_LEVEL = 18,
+    ICCP_DBG_CNTR_MSG_STP_INSTANCE_PRIORITY = 19,
+    ICCP_DBG_CNTR_MSG_STP_CONFIGURATION_DIGEST = 20,
+    ICCP_DBG_CNTR_MSG_STP_TC_INSTANCES = 21,
+    ICCP_DBG_CNTR_MSG_STP_ROOT_TIME_PARAM = 22,
+    ICCP_DBG_CNTR_MSG_STP_MIST_ROOT_TIME_PARAM = 23,
+    ICCP_DBG_CNTR_MSG_STP_SYNC_REQ     = 24,
+    ICCP_DBG_CNTR_MSG_STP_SYNC_DATA    = 25,
+    ICCP_DBG_CNTR_MSG_STP_PO_PORT_MAP  = 26,
+    ICCP_DBG_CNTR_MSG_STP_AGE_OUT      = 27,
+    ICCP_DBG_CNTR_MSG_STP_COMMON_MSG   = 28,
+    ICCP_DBG_CNTR_MSG_MAX
+};
+typedef enum ICCP_DBG_CNTR_MSG ICCP_DBG_CNTR_MSG_e;
+
+/* Count messages sent to MCLAG peer */
+#define MLACP_SET_ICCP_TX_DBG_COUNTER(csm, tlv_type, status)\
+do{\
+    ICCP_DBG_CNTR_MSG_e dbg_type;\
+    dbg_type = mlacp_fsm_iccp_to_dbg_msg_type(tlv_type);\
+    if (csm && ((dbg_type) < ICCP_DBG_CNTR_MSG_MAX) && ((status) < ICCP_DBG_CNTR_STS_MAX))\
+        ++MLACP(csm).dbg_counters.iccp_counters[dbg_type][ICCP_DBG_CNTR_DIR_TX][status];\
+}while(0);
+
+/* Count messages received from MCLAG peer */
+#define MLACP_SET_ICCP_RX_DBG_COUNTER(csm, tlv_type, status)\
```

Comment:
The difference between the macro MLACP_SET_ICCP_RX_DBG_COUNTER() and MLACP_SET_ICCP_TX_DBG_COUNTER() is only on the direction.  I would sugggest that you pass that as an additional parameter (ICCP_DBG_CNTR_DIR_TX/ICCP_DBG_CNTR_DIR_RX) and only have one single MACRO instead of repeated code...